.grit/patterns/mui4_to_mui5.md

```diff
@@ -1,21 +1,71 @@
 # Upgrade MUI v4 to MUI v5
 
 This migration handles some of the cases not covered in the [official codemod](https://mui.com/material-ui/migration/migration-v4/):
+
 - Renames `theme.palette.type` to `theme.palette.mode`
 
-tags: #react, #migration, #complex, #alpha, #hidden
+- Changes `ThemeProvider` import from `@mui/styles` to `@mui/material/styles`
+
+
+- Changes default theme.palette.info colors from `cyan[300]` to `lightBlue[500]`
+- Changes default theme.palette.info colors from `cyan[500]` to `lightBlue[700]`
+- Changes default theme.palette.info colors from `cyan[700]` to `lightBlue[900]`
+
+- Changes default theme.palette.success colors from `green[300]` to `green[500]`
+- Changes default theme.palette.success colors from `green[500]` to `green[700]`
+- Changes default theme.palette.success colors from `green[700]` to `green[900]`
+
+tags: #react, #migration, #complex, #alpha, #hidden #mui
+
 
 ```grit
 engine marzano(0.1)
 language js
 
-`createTheme($theme)` where {
-  $theme <: contains `palette: { $palette }`,
-  $palette <: contains `type: $arg` => `mode: $arg`
+pattern rename_palette_type () {
+  `createTheme($theme)` where {
+    $theme <: contains `palette: { $palette }`,
+    $palette <: contains `type: $arg` => `mode: $arg`
+  }
+}
+
+pattern replace_theme_provider_import () {
+   `ThemeProvider` as $target where {
+    $target <: replace_import(old= `'@mui/styles'`, new=`'@mui/material/styles'`),
+  }
+}
+
+pattern upgrade_info_palette () {
+    `main: $color[$value]` where {
+          $color <: r"cyan" => `lightBlue`,
+      or {
+          $value <: `300` => `500`,
+          $value <: `500` => `700`,
+          $value <: `700` => `900`
+      }
+  }
+}
+
+pattern upgrade_success_palette () {
+  `main: $color[$value]` where {
+      $color <: r"green" => `green`,
+    or {
+      $value <: `300` => `500`,
+      $value <: `500` => `800`,
+      $value <: `700` => `900`
+    }
+  }
+}
```

Comment:
These patterns are overly broad. We should only be upgrading colors in the context of MUI.